packages/react-native-bpk-component-calendar/src/ios/CalendarBridge/RCTBPKCalendarManager.m

```diff
@@ -20,23 +20,28 @@
 #import <React/RCTBridge.h>
 #import <React/RCTUIManager.h>
 
+#import <Backpack/Color.h>
+
 #import "RCTBPKCalendar.h"
 #import "RCTBPKCalendarDateUtils.h"
 #import "RCTConvert+RCTBPKCalendar.h"
+#import "RCTBPKColorBucket.h"
+#import "RCTBPKDateMatcher.h"
 
 @interface RCTBPKCalendarManager () <BPKCalendarDelegate>
-
+@property(nonatomic, strong) NSArray<RCTBPKColorBucket *> *colorBuckets;
+@property(nonatomic, strong) RCTBPKCalendar *calendar;
```

Comment:
The data should live either in the view instance itself or in the shadow view. For example [`RCTTextShadowView`](https://github.com/facebook/react-native/blob/86cad7d69099baaaeae10ac7c0912a023101f257/Libraries/Text/Text/RCTTextShadowView.h#L14) keeps data within it that is later used as part of layout. Since our case doesn't affect the layout or Yoga I think storing the buckets inside `RCTBPKCalendar` is the right choice here

Reply:
This will require a change to this delegate, right?
Because if we don't have a reference to the Calendar we would have to be told by the delegate method what calendar it relates to...
And then cast the calendar to a RCTBPKCalendar to get the colorBuckets?

https://github.com/Skyscanner/backpack-ios/blob/master/Backpack/Calendar/Classes/BPKCalendar.h#L76-L81

Reply:
Yes 👍 All delegate methods should have `calendar:` as the first argument by convention anyway. We can keep those old methods and deprecate them to avoid a breakign change